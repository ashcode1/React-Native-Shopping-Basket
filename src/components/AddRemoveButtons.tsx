import React, { FC } from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { SquareButton, TextButton } from './';

interface Props {
  quantity: number;
  addOnPress: () => void;
  subtractOnPress: () => void;
  removeAllOnPress: () => void;
}

const AddRemoveButtons: FC<Props> = (props) => {
  const { quantity, addOnPress, subtractOnPress, removeAllOnPress } = props;
  const { containerStyles, buttonsContainerStyles } = styles;
  return (
    <View style={containerStyles}>
      <View style={buttonsContainerStyles}>
        <SquareButton name="md-remove" onPress={subtractOnPress} />
        <Text>{quantity}</Text>
        <SquareButton name="md-add" onPress={addOnPress} />
      </View>
      <TextButton text="remove" onPress={removeAllOnPress} />
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyles: {
    width: 95,
    marginLeft: 15,
    justifyContent: 'flex-end',
  },
  buttonsContainerStyles: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 10,
  },
});

export { AddRemoveButtons };
