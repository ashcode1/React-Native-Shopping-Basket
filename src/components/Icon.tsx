import React, { FC } from 'react';
import { Icon as NBIcon } from 'native-base';
import { StyleSheet, View } from 'react-native';

interface Props {
  name: string;
  style: object;
}

const Icon: FC<Props> = (props) => {
  const { name, style } = props;
  const { containerStyles, iconStyles } = styles;
  return (
    <View style={containerStyles}>
      <NBIcon name={name} style={[iconStyles, style]} />
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyles: {
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    height: 25,
  },
  iconStyles: {
    fontSize: 16,
    textAlign: 'center',
    alignSelf: 'center',
  },
});

export { Icon };
