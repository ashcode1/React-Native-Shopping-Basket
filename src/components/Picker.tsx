import React, { useState, FC } from 'react';
import { Text, StyleSheet } from 'react-native';
import DropDownPicker from 'react-native-dropdown-picker';
import { SelectedColor } from '../interfaces';
import { lightGrey } from '../theme/colors';

interface Props {
  defaultValue: string;
  onChangeItem: (selectedColor: SelectedColor) => void;
}

const Picker: FC<Props> = (props) => {
  const { defaultValue, onChangeItem } = props;
  const {
    mainStyles,
    containerStyles,
    itemStyles,
    dropDownStyles,
    textStyles,
  } = styles;
  return (
    <>
      <Text style={textStyles}>Filter by colour:</Text>
      <DropDownPicker
        items={[
          { label: 'stone', value: 'Stone' },
          { label: 'black', value: 'Black' },
          { label: 'red', value: 'Red' },
          { label: 'none', value: 'None' },
        ]}
        defaultValue={defaultValue}
        containerStyle={containerStyles}
        style={mainStyles}
        itemStyle={itemStyles}
        dropDownStyle={dropDownStyles}
        onChangeItem={onChangeItem}
      />
    </>
  );
};

const styles = StyleSheet.create({
  mainStyles: {
    backgroundColor: lightGrey,
  },
  containerStyles: {
    height: 40,
    width: 95,
    marginLeft: 15,
  },
  itemStyles: {
    justifyContent: 'flex-start',
  },
  dropDownStyles: {
    backgroundColor: lightGrey,
  },
  textStyles: {
    marginLeft: 15,
    marginVertical: 5,
  },
});

export { Picker };
