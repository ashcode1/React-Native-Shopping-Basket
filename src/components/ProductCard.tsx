import React, { FC } from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';
import { AddRemoveButtons } from './';
import { lightGrey } from '../theme/colors';

interface Props {
  id: number;
  name: string;
  price: number;
  image: string;
  quantity: number;
  addOnPress: () => void;
  subtractOnPress: () => void;
  removeAllOnPress: () => void;
}

const ProductCard: FC<Props> = (props) => {
  const {
    image,
    name,
    price,
    quantity,
    addOnPress,
    subtractOnPress,
    removeAllOnPress,
  } = props;
  const {
    containerStyles,
    imageStyles,
    labelsContainerStyles,
    priceStyles,
    lineStyles,
  } = styles;
  return (
    <>
      <View style={containerStyles}>
        <Image
          resizeMode="contain"
          style={imageStyles}
          source={{
            uri: image,
          }}
        />
        <View style={labelsContainerStyles}>
          <Text>{name}</Text>
          <Text style={priceStyles}>{`Â£${price}`}</Text>
        </View>
        <AddRemoveButtons
          quantity={quantity}
          addOnPress={addOnPress}
          subtractOnPress={subtractOnPress}
          removeAllOnPress={removeAllOnPress}
        />
      </View>
      <View style={lineStyles} />
    </>
  );
};

const styles = StyleSheet.create({
  containerStyles: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    padding: 15,
  },
  imageStyles: {
    alignSelf: 'flex-start',
    marginRight: 15,
    height: 150,
    width: 95,
  },
  labelsContainerStyles: {
    flex: 1,
  },
  priceStyles: {
    fontWeight: 'bold',
    marginTop: 5,
  },
  lineStyles: {
    flexDirection: 'row',
    borderWidth: 1,
    borderColor: lightGrey,
  },
});

export { ProductCard };
