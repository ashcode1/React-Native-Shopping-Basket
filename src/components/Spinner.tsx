import React, { FC } from 'react';
import { Spinner as NBSpinner } from 'native-base';
import { StyleSheet, Dimensions, View } from 'react-native';

const Spinner: FC = () => {
  const { containerStyles } = styles;
  return (
    <View style={containerStyles}>
      <NBSpinner color="black" />
    </View>
  );
};

const styles = StyleSheet.create({
  containerStyles: {
    height: Dimensions.get('screen').height,
    justifyContent: 'center',
  },
});

export { Spinner };
