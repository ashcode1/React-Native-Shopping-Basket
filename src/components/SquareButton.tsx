import React, { FC } from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { Icon } from './';
import { black } from '../theme/colors';

interface Props {
  onPress: () => void;
  name: string;
}

const SquareButton: FC<Props> = (props) => {
  const { onPress, name } = props;
  const { containerStyles, iconStyles } = styles;
  return (
    <TouchableOpacity onPress={onPress} style={containerStyles}>
      <Icon name={name} style={iconStyles} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  containerStyles: {
    width: 25,
    height: 25,
    backgroundColor: black,
  },
  iconStyles: {
    color: 'white',
  },
});

export { SquareButton };
