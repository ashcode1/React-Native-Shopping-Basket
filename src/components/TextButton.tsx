import React, { FC } from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

interface Props {
  onPress: () => void;
  text: string;
}

const TextButton: FC<Props> = (props) => {
  const { onPress, text } = props;
  const { containerStyles } = styles;
  return (
    <TouchableOpacity onPress={onPress} style={containerStyles}>
      <Text>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  containerStyles: {
    alignSelf: 'center',
  },
});

export { TextButton };
