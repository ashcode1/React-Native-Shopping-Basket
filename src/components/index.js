export * from './AddRemoveButtons';
export * from './Icon';
export * from './Picker';
export * from './ProductCard';
export * from './Spinner';
export * from './SquareButton';
export * from './TextButton';
export * from './TotalAmount';
