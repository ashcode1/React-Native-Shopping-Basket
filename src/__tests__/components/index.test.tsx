import * as index from '../../components/index';

describe('Components Index file', () => {
  it('Remains Unchanged', () => {
    expect(index).toMatchSnapshot();
  });
});
