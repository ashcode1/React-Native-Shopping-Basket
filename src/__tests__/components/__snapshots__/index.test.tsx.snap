// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Index file Remains Unchanged 1`] = `
Object {
  "AddRemoveButtons": [Function],
  "Icon": [Function],
  "Picker": [Function],
  "ProductCard": [Function],
  "Spinner": [Function],
  "SquareButton": [Function],
  "TextButton": [Function],
  "TotalAmount": [Function],
}
`;
